src/lib/fileHelpers.ts

```diff
@@ -61,13 +75,78 @@ export const convertFolderToModule = (
             );
 
             edit.renameFile(e.files[0], vscode.Uri.file(uniqueModulePath));
-
             vscode.workspace.applyEdit(edit).then(() => {
               copySampleModuleFilesToFolder(uniqueModulePath);
               resolve(edit);
             });
           }
+          reject(edit);
+        } catch (e: any) {
+          reject(e);
+        }
+      })
+    );
+  };
+};
+
+/*
+ * Returns a non-duplicate folder name with the format <folderName>.functions
+ * It will add a number to the end of the folder name if it already exists
+ * @param folderName - name of a folder with or without a .functions extension
+ */
+const getUniqueFunctionsFolderName = (folderPath: string) => {
```

Comment:
This function (`getUniqueFunctionsFolderName`) and `convertFolderToServerlessFunction` are super similar to the module folder generation code, to the extent that we could generalize and make it a little DRYer, but I'm not sure it would make the code any more readable and these should be the only two cases we need to use this logic so not a ton of benefit to it.

Reply:
Actually changed my mind on this overnight, I think `getUniqueFunctionsFolderName` and `getUniqueModulesFolderName` should be generalized at least, other two are ok to have separate.